source/file_layout.rst

```diff
@@ -169,24 +169,39 @@ Each block begins with the following header:
   bytes.
 
 - ``used_size`` (64-bit unsigned integer, big-endian): The amount of
-  used space for the block (not including the header), in bytes.
+  used space for the block on disk (not including the header), in
+  bytes.
+
+- ``memory_size`` (64-bit unsigned integer, big-endian): The size of
+  the block when decoded, in bytes.  If ``encoding`` is all zeros
+  (indicating no encoding), it **must** be equal to ``used_size``.  If
+  there is an encoding, this is the size of the decoded block data,
+  which **may** be different from the size of the encoded block in the
+  ASDF file.
```

Comment:
Yeah -- I'm not crazy about the name either.  Just couldn't come up with a better name.  `actual_size` is fine.  `data_size` might be ok also.